src/compile/selection/interval.ts

```diff
@@ -156,29 +125,40 @@ export function projections(selCmpt: SelectionComponent) {
   return {x, xi, y, yi};
 }
 
-function channelSignal(model: UnitModel, selCmpt: SelectionComponent, channel: Channel): any {
-  const name  = channelSignalName(selCmpt, channel),
+function channelSignals(model: UnitModel, selCmpt: SelectionComponent, channel: Channel): any {
+  const vname = channelSignalName(selCmpt, channel, 'visual'),
+      dname = channelSignalName(selCmpt, channel, 'data'),
+      hasScales = scales.has(selCmpt),
+      scaleName = model.scaleName(channel),
+      scaleStr = stringValue(scaleName),
+      scaleType = model.getComponent('scales', channel).type,
       size  = model.getSizeSignalRef(channel === X ? 'width' : 'height').signal,
-      coord = `${channel}(unit)`,
-      invert = invertFn.bind(null, model, selCmpt, channel);
-
-  return {
-    name: name,
-    value: [],
-    on: scales.has(selCmpt) ? [] : events(selCmpt, function(on: any[], evt: any) {
-      on.push({
-        events: evt.between[0],
-        update: invert(`[${coord}, ${coord}]`)
-      });
+      coord = `${channel}(unit)`;
 
-      on.push({
-        events: evt,
-        update: `[${name}[0], ` + invert(`clamp(${coord}, 0, ${size})`) + ']'
-      });
+  const on = events(selCmpt, function(def: any[], evt: any) {
```

Comment:
Can you add a type? Even if just `type VgEventStream = any`. Then this question will never come up again.